### PR TITLE
Create new commands for audition

### PR DESCRIPTION
**Pull request by Evgenij-Kovpak**

### CODE REVIEW

File: Elevel/Application/Features/AuditionCommands/DeleteAuditionCommand.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+                if (audition is null)
+                {
+                    return null;
+                }
+                audition.Deleted = true;
+                await _context.SaveChangesAsync(cancellationToken);
+                return new Response
+                {
+                    Id = audition.Id,
+                    Deleted = audition.Deleted
```
Comment (parys) [inline Elevel/Application/Features/AuditionCommands/DeleteAuditionCommand.cs:41]: don't need in response, just Id

File: Elevel/Application/Features/AuditionCommands/DeleteAuditionCommand.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                _context = context;
+                _mapper = mapper;
+            }
+
+            public async Task<Response> Handle(Request request, CancellationToken cancellationToken)
+            {
+                var audition = await _context.Auditions.FirstOrDefaultAsync(x => x.Id == request.Id, cancellationToken);
+                if (audition is null)
+                {
+                    return null;
```
Comment (parys) [inline Elevel/Application/Features/AuditionCommands/DeleteAuditionCommand.cs:34]: should be consistent, there are many places with throw new NotFoundException

File: Elevel/Application/Features/AuditionCommands/GetAuditionDetailQuery.cs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+            public Handler(IApplicationDbContext context, IMapper mapper)
+            {
+                _context = context;
+                _mapper = mapper;
+            }
+
+            public async Task<Response> Handle(Request request, CancellationToken cancellationToken)
+            {
+                var audition = await _context.Auditions
+                    .Include(x=>x.Questions).ThenInclude(i => i.Answers)
```
Comment (parys) [inline Elevel/Application/Features/AuditionCommands/GetAuditionDetailQuery.cs:38]: please map here with ProjectTo and you can omit Include and ThenInclude

File: Elevel/Persistence/Configurations/AuditionConfiguration.cs [hunk 4]
```diff
@@ -13,8 +13,6 @@ public void Configure(EntityTypeBuilder<Audition> builder)
             builder.Property(x => x.CreationDate)
                 .HasDefaultValueSql("SYSDATETIMEOFFSET()");
 
-            builder.HasQueryFilter(x => !x.Deleted);
```
Comment (parys) [inline Elevel/Persistence/Configurations/AuditionConfiguration.cs:16]: why?

File: Elevel/Persistence/Configurations/QuestionConfiguration.cs [hunk 5]
```diff
@@ -21,7 +21,9 @@ public void Configure(EntityTypeBuilder<Question> builder)
             builder.Property(x => x.CreationDate)
                 .HasDefaultValueSql("SYSDATETIMEOFFSET()");
 
-            builder.HasQueryFilter(x => !x.Deleted);
```
Comment (parys) [inline Elevel/Persistence/Configurations/QuestionConfiguration.cs:24]: why?

File: Elevel/Persistence/Configurations/TopicConfiguration.cs [hunk 6]
```diff
@@ -15,8 +15,6 @@ public void Configure(EntityTypeBuilder<Topic> builder)
 
             builder.Property(x => x.CreationDate)
                 .HasDefaultValueSql("SYSDATETIMEOFFSET()");
-
-            builder.HasQueryFilter(x => !x.Deleted);
```
Comment (parys) [inline Elevel/Persistence/Configurations/TopicConfiguration.cs:19]: why?

File: Elevel/Application/Features/AuditionCommands/DeleteAuditionCommand.cs (hunk 1 above)
Comment (rinchan603) [reply Elevel/Application/Features/AuditionCommands/DeleteAuditionCommand.cs:41]: Done

File: Elevel/Elevel.Api/appsettings.json [hunk 7]
```diff
@@ -9,7 +9,8 @@
 
   "AllowedHosts": "*",
   "ConnectionStrings": {
-    "DefaultConnection": "Server=localhost;Database=Elevel;Trusted_Connection=True;MultipleActiveResultSets=true",
+    "DefaultConnection": "Server=PC\\SQLEXPRESS;Database=Elevel;Trusted_Connection=True;MultipleActiveResultSets=true",
+    //"DefaultConnection": "Server=localhost;Database=Elevel;Trusted_Connection=True;MultipleActiveResultSets=true",
```
Comment (parys) [inline Elevel/Elevel.Api/appsettings.json:13]: please do not change connection string again and again

File: Elevel/Application/Features/AuditionCommands/GetAuditionDetailQuery.cs (hunk 3 above)
Comment (rinchan603) [reply Elevel/Application/Features/AuditionCommands/GetAuditionDetailQuery.cs:38]: Done

File: Elevel/Elevel.Api/appsettings.json (hunk 7 above)
Comment (rinchan603) [reply Elevel/Elevel.Api/appsettings.json:13]: Ok

File: Elevel/Application/Features/AuditionCommands/DeleteAuditionCommand.cs (hunk 2 above)
Comment (rinchan603) [reply Elevel/Application/Features/AuditionCommands/DeleteAuditionCommand.cs:34]: Done

**Review by parys — APPROVED** [review]

**Review by parys — APPROVED** [review]

**Comment by rinchan603** [comment]
[BE/Iryna Kazlova/Kowpak Evgeniy/Create new commands for audition #119](https://trello.com/c/Oen6qW51/174-be-iryna-kazlova-kowpak-evgeniy-create-new-commands-for-audition-119)

**Review by andreyrogov8 — APPROVED** [review]

**Pull request merged**